Unittests

philomath213 opened the pull request:
see #3

philomath213 committed:
add LSBExtractor unittests

philomath213 committed:
fix imports for stegano module

philomath213 committed:
refactore lsb_extractor.py

philomath213 committed:
fix unittests

philomath213 committed:
add zxing decoder unittest

philomath213 committed:
add TestLcg

philomath213 committed:
remove old test files

philomath213 committed:
fix crypto imports

youben11 commented on tests/tests.py:
```diff
@@ -0,0 +1,58 @@
+        self.assertEqual(msg, "LSB cycle hidden message")
+
+
+class TestZxing(unittest.TestCase):
```
does the tests here works for you?

philomath213 replied:
the test should work on both python versions 2 and 3, but the "Zxing" module use "HTMLParser" module which is available on python2 only.
we should re-implement the "Zxing" module and use "lxml" module to parse the html.

youben11 replied:
on it

youben11 committed:
update zxing

HTMLParser was not longer available on python3, necessity to use another
similar package

youben11 committed:
Merge pull request #5 from sudo-root-team/update-zxing

update zxing

youben11 committed:
get str response

youben11 commented on tests/tests.py:
```diff
@@ -0,0 +1,58 @@
+
+    def test_lcg(self):
+        lcg = Lcg(0x66e158441b6995, 0xB, 1 << 85, 53)
+        state = lcg.get_state(2752470789, 3367609997, 1185935283)
```
this one is also failing, need an update to the LCG module

philomath213 replied:
now it's ready to be merged

youben11 committed:
update lcg: python3 integer division

philomath213 committed:
Merge pull request #6 from sudo-root-team/update-lcg

update lcg: python3 integer division
